src/h3lib/lib/algos.c

```diff
@@ -118,7 +119,7 @@ void H3_EXPORT(kRingDistances)(H3Index origin, int k, H3Index* out,
         // Fast algo failed, fall back to slower, correct algo
         // and also wipe out array because contents untrustworthy
         for (int i = 0; i < maxIdx; i++) {
-            out[i] = 0;
+            out[i] = H3_INVALID_INDEX;
             distances[i] = 0;
```

Comment:
Not required, but it's worth noting that `0` is actually a valid value in `distances` - maybe worth defining `INVALID_DISTANCE = -1` or similar?